sycl/doc/design/DeviceConfigFile.md

```diff
@@ -0,0 +1,332 @@
+def AspectAccelerator : Aspect<"accelerator">;
+def AspectCustom : Aspect<"custom">;
+def AspectFp16 : Aspect<"fp16">;
+def AspectFp64 : Aspect<"fp64">;
+def AspectImage : Aspect<"image">;
+def AspectOnline_compiler : Aspect<"online_compiler">;
+def AspectOnline_linker : Aspect<"online_linker">;
+def AspectQueue_profiling : Aspect<"queue_profiling">;
+def AspectUsm_device_allocations : Aspect<"usm_device_allocations">;
+def AspectUsm_host_allocations : Aspect<"usm_host_allocations">;
+def AspectUsm_shared_allocations : Aspect<"usm_shared_allocations">;
+def AspectUsm_system_allocations : Aspect<"usm_system_allocations">;
+def AspectExt_intel_pci_address : Aspect<"ext_intel_pci_address">;
+def AspectExt_intel_gpu_eu_count : Aspect<"ext_intel_gpu_eu_count">;
+def AspectExt_intel_gpu_eu_simd_width : Aspect<"ext_intel_gpu_eu_simd_width">;
+def AspectExt_intel_gpu_slices : Aspect<"ext_intel_gpu_slices">;
+def AspectExt_intel_gpu_subslices_per_slice : Aspect<"ext_intel_gpu_subslices_per_slice">;
+def AspectExt_intel_gpu_eu_count_per_subslice : Aspect<"ext_intel_gpu_eu_count_per_subslice">;
+def AspectExt_intel_max_mem_bandwidth : Aspect<"ext_intel_max_mem_bandwidth">;
+def AspectExt_intel_mem_channel : Aspect<"ext_intel_mem_channel">;
+def AspectUsm_atomic_host_allocations : Aspect<"usm_atomic_host_allocations">;
+def AspectUsm_atomic_shared_allocations : Aspect<"usm_atomic_shared_allocations">;
+def AspectAtomic64 : Aspect<"atomic64">;
+def AspectExt_intel_device_info_uuid : Aspect<"ext_intel_device_info_uuid">;
+def AspectExt_oneapi_srgb : Aspect<"ext_oneapi_srgb">;
+def AspectExt_oneapi_native_assert : Aspect<"ext_oneapi_native_assert">;
+def AspectHost_debuggable : Aspect<"host_debuggable">;
+def AspectExt_intel_gpu_hw_threads_per_eu : Aspect<"ext_intel_gpu_hw_threads_per_eu">;
+def AspectExt_oneapi_cuda_async_barrier : Aspect<"ext_oneapi_cuda_async_barrier">;
+def AspectExt_oneapi_bfloat16_math_functions : Aspect<"ext_oneapi_bfloat16_math_functions">;
+def AspectExt_intel_free_memory : Aspect<"ext_intel_free_memory">;
+def AspectExt_intel_device_id : Aspect<"ext_intel_device_id">;
+def AspectExt_intel_memory_clock_rate : Aspect<"ext_intel_memory_clock_rate">;
+def AspectExt_intel_memory_bus_width : Aspect<"ext_intel_memory_bus_width">;
+def AspectEmulated : Aspect<"emulated">;
+    
+def TargetTable : DynamicTable { 
+    let FilterClass = "TargetInfo";
+    let Fields = ["TargetName", "aspects", "maySupportOtherAspects",
+                  "subGroupSizes", "aotToolchain", "aotToolchain-options"];
+    string TypeOf_aspects = "list<Aspect>";
+}
+
+class TargetInfo <string tgtName, list<Aspect> aspectList, bit otherAspects,
+                  list<int> listSubGroupSizes, string toolchain, string option>
+{
+    list<Aspect> aspects = aspectList;
+    bits<1> maySupportOtherAspects = otherAspects;
+    list<int> subGroupSizes = listSubGroupSizes;
+    string aotToolchain = toolchain;
+    string aotToolchain-options = option;
```

Comment:
It's unclear to me how we are going to be using this value, but if it's just something that we will simply pass on to the aot command, couldn't we simply add as many options as required in a single string? Like `string aotToolchainOptions = -option1 val1, -option2 val2`. Is there any scenario where we would need different strings?

Reply:
Good point. I don't think we really want to parse AOT toolchain options and can store them as a single string. In that case, however, `.td` file should contain something like `-device skl` instead of just `skl`.